app/serializers/participant_serializer.rb

```diff
@@ -30,10 +30,8 @@ class ParticipantSerializer
   end
 
   attributes :participant_type do |user|
-    if user.early_career_teacher_profile
-      :ect
-    else
-      :mentor
+    if participant_active?(user)
```

Comment:
Now this is being repeated lots (and there seems to be use of the plural form when only one attribute is being listed, would it not be better to create a private method to wrapper it? 

e.g. 
```
class << self
 def active_participant_attribute(attr, &blk)
    attribute attr do |user|
      if participant_active?(user)
        blk.call
      end
    end
  end
end
```

then you could dry up the code with:
```
active_participant_attribute :email do |user|
  user.email
end

active_participant_attribute :full_name do |user|
    user.full_name
end
```

Reply:
I like this, but can't get it working, some variation on 

```
NoMethodError:
  undefined method `active_participant_attribute' for ParticipantSerializer:Class
```

I'm going to go as-is for now, then come back this this if I have time

Reply:
Did you miss the "class << self"? I did a test refactor on it, and it went from 55 lines to .......

55 lines. :)